torch/csrc/distributed/c10d/c10d_frontend.cpp

```diff
@@ -31,4 +48,105 @@ void DistributedC10d::set_default_pg_init_method(
   default_pg_init_method_ = init_method;
 }
 
+std::shared_ptr<ProcessGroup> DistributedC10d::newProcessGroupHelper(
+    const int64_t world_size,
+    const int64_t rank,
+    const std::vector<int64_t>& group_ranks,
```

Comment:
Shouldn't we have some unit tests covering this new API? My proposal would be to actually modify our current code in `distributed_c10d.py` to use this and as a result all of our existing tests would cover it.

Reply:
Hi Pritam, great point! Soon, we plan to reuse all existing tests that involve torch.distributed by adding a test configuration that toggles which c10d frontend the test uses. This way we ensure the two frontends remain consistent. That's also why we don't have new test cases just for new implementation for now.